app/repository/RiderStageConnectionRepositoryImpl.java

```diff
@@ -25,55 +25,79 @@ public RiderStageConnectionRepositoryImpl(JPAApi jpaApi, DatabaseExecutionContex
     }
 
     @Override
-    public CompletionStage<Stream<RiderStageConnection>> getAllRiderStageConnections() {
-        return supplyAsync(() -> wrap (this::getAllRiderStageConnections), databaseExecutionContext);
+    public CompletionStage<Stream<RiderStageConnection>> getAllRiderStageConnections(long stageId) {
+        return supplyAsync(() -> wrap (entityManager -> getAllRiderStageConnections(entityManager, stageId)), databaseExecutionContext);
     }
 
-    private Stream<RiderStageConnection> getAllRiderStageConnections(EntityManager em){
-        List<RiderStageConnection> riderStageConnections = em.createQuery("select rSC from RiderStageConnection rSC", RiderStageConnection.class).getResultList();
-        return riderStageConnections.stream();
+    private Stream<RiderStageConnection> getAllRiderStageConnections(EntityManager em, long stageId){
+        TypedQuery<RiderStageConnection> query = em.createQuery("select rSC from RiderStageConnection rSC where rSC.stage.id = :stageId" , RiderStageConnection.class);
+        query.setParameter("stageId", stageId);
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "stageId" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)